src/doc_table.h

```diff
@@ -44,19 +62,19 @@ typedef struct {
 DocTable NewDocTable(size_t cap);
 
 /* Get the metadata for a doc Id from the DocTable.
-*  If docId is not inside the table, we return NULL */
+ *  If docId is not inside the table, we return NULL */
 RSDocumentMetadata *DocTable_Get(DocTable *t, t_docId docId);
 
 /* Put a new document into the table, assign it an incremental id and store the metadata in the
-* table.
-*
-* NOTE: Currently there is no deduplication on the table so we do not prevent dual insertion of the
-* same key. This may result in document duplication in results  */
-t_docId DocTable_Put(DocTable *t, const char *key, double score, u_char flags, const char *payload,
-                     size_t payloadSize);
+ * table.
+ *
+ * NOTE: Currently there is no deduplication on the table so we do not prevent dual insertion of the
+ * same key. This may result in document duplication in results  */
+t_docId DocTable_Put(DocTable *t, RSDocumentKey key, double score, u_char flags,
+                     const char *payload, size_t payloadSize);
 
 /* Get the "real" external key for an incremental id. Returns NULL if docId is not in the table. */
-const char *DocTable_GetKey(DocTable *t, t_docId docId);
+RSDocumentKey DocTable_GetKey(DocTable *t, t_docId docId);
```

Comment:
I'd add a check macro like `RSDOCKEY_ISNULL(k)` for functions calling this, but right now there aren't any outside the tests...